pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -142,6 +142,11 @@ func (m *kubeGenericRuntimeManager) getImageUser(image string) (*int64, string,
 // isInitContainerFailed returns true if container has exited and exitcode is not zero
 // or is in unknown state.
 func isInitContainerFailed(status *kubecontainer.Status) bool {
+	// When oomkilled occurs, init container should be considered as a failure.
+	if status.Reason == "OOMKilled" {
```

Comment:
The Reason here should come from runtime, I want to know when the container is killed due to oom,
all kinds of runtime reporting reason is oomkilled?
/cc @249043822

Reply:
ok， I will ensure other runtime behave.

Reply:
I think that you should discuss this with sig-node, I'm not familiar with this function but the description says
> // isInitContainerFailed returns true if container has exited and exitcode is not zero
> // or is in unknown state.

the change you are adding is clearly not in that scope, I don't know if and OOMKilled container can be considered as a failure honestly,

Reply:
This change should be approved by sig-node whether we should track non-init process(child process) OOM kills. Maybe there are apps that can tolerate OOM kills of non init processes. @ehashman

Reply:
I think the answer here is "it depends" because usually OOM kills result in an exit code of 137.
POSIX says that 0 means success, but I have doubts most init containers can happily finish their job to completion after being killed, even if they could handle it:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/docker_container.go#L414-L415

I would say having an initContainer OOMKilled isn't very encouraging to let this pod run successfully, but it's only me.

Reply:
@matthyx
I agree with you. In general, oomkilled is an exception.

Reply:
then update the description of the function please,

Reply:
@aojea ok, I will do as you see. Thanks very much again.